### PR TITLE
Add org.aliflang.lang

### PR DESCRIPTION
**Pull request by sulimanp**

### CODE REVIEW

**Comment by sulimanp** [comment]
bot, build org.aliflang.lang

File: org.aliflang.lang.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
+app-id: org.aliflang.lang
+runtime: org.freedesktop.Sdk
+runtime-version: '19.08'
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.yml:3]: Any reason for not bumping this to 20.08?

Comment (sulimanp) [reply org.aliflang.lang.yml:3]: yes, I can't build webkitgtk-2.4.11 with 20.08 but build successful with 19.08
I'll rebuild it and send you the error

Comment (sulimanp) [reply org.aliflang.lang.yml:3]:
```
/usr/include/unicode/unorm.h:218:1: note: declared here
  218 | unorm_normalize(const UChar *source, int32_t sourceLength,
      | ^~~~~~~~~~~~~~~
  CXX      Source/WebCore/platform/image-decoders/webp/libPlatform_la-WEBPImageDecoder.lo
Source/WebCore/platform/graphics/SegmentedFontData.cpp: In member function ‘virtual bool WebCore::SegmentedFontData::containsCharacters(const UChar*, int) const’:
Source/WebCore/platform/graphics/SegmentedFontData.cpp:65:9: error: expected ‘;’ before ‘if’
   65 |         if (!containsCharacter(c))
      |         ^~
make[1]: *** [GNUmakefile:51072: Source/WebCore/platform/graphics/libPlatform_la-SegmentedFontData.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/run/build/webkitgtk'
make: *** [GNUmakefile:25837: all] Error 2
Error: module webkitgtk: Child process exited with code 2

```

**Comment by sulimanp** [comment]
bot, build org.aliflang.lang

**Comment by sulimanp** [comment]
bot, build org.aliflang.lang

File: org.aliflang.lang.desktop [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.desktop:1]: This should be upstreamed as it's not Flathub specific

File: org.aliflang.lang.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+	<id>org.aliflang.lang</id>
+	<content_rating type="oars-1.0" />
+	<project_license>GPL-3.0 and LGPL-3.0</project_license>
+	<metadata_license>CC-BY-SA-3.0</metadata_license>
+	<name>Alif Programming Language</name>
+	<summary>لغة ألف, ألف ستوديو, لغة البرمجة العربية, بيئة تطوير, تعليم</summary>
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.metainfo.xml:8]: I would write the summary & description in english & translate them in arabic. The metainfo file can be translated either automatically using gettext at build stage or you can add the translations manually to the file yourself.

File: org.aliflang.lang.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.metainfo.xml:2]: The metainfo should be upstream as well

File: org.aliflang.lang.yml [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+sdk: org.freedesktop.Sdk
+command: alifstudio
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --filesystem=home
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.yml:9]:
```suggestion
```
you have this in filesystem=host already

File: org.aliflang.lang.yml [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+  - --filesystem=host
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.yml:10]: What does it need host access for?

File: org.aliflang.lang.yml [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+  - name: desktop-files
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/share/icons/hicolor/32x32/apps/ /app/share/metainfo/ /app/share/applications/
+      - install "org.aliflang.lang.png" /app/share/icons/hicolor/32x32/apps/org.aliflang.lang.png
+      - install "org.aliflang.lang.metainfo.xml" /app/share/metainfo/
+      - install "org.aliflang.lang.desktop" /app/share/applications/
+      - ln -s /app/share/applications /app/usr/share/applications
+      - ln -s /app/share/appdata /app/usr/share/appdata
+      - ln -s /app/share/icons /app/usr/share/icons
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.yml:141]:
```suggestion
```
you don't need those

File: org.aliflang.lang.yml [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+        url: https://github.com/aliflanguage/alifstudio.git
+        branch: flatpak-fork-unstable
+        commit: 9e042b8dfe176065c2c7a2866ed105d7ee5ad6e3
+
+
+  - name: desktop-files
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/share/icons/hicolor/32x32/apps/ /app/share/metainfo/ /app/share/applications/
+      - install "org.aliflang.lang.png" /app/share/icons/hicolor/32x32/apps/org.aliflang.lang.png
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.yml:136]: Flathub requires a 128px icon or a scalable one (SVG)

File: org.aliflang.lang.yml [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+      - install "org.aliflang.lang.metainfo.xml" /app/share/metainfo/
+      - install "org.aliflang.lang.desktop" /app/share/applications/
+      - ln -s /app/share/applications /app/usr/share/applications
+      - ln -s /app/share/appdata /app/usr/share/appdata
+      - ln -s /app/share/icons /app/usr/share/icons
+    sources:
+      - type: file
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.yml:143]: The icon should be upstreamed as well

File: org.aliflang.lang.yml [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+      - g++ -o "bin/alifstudio" "alifstudio.o" -L"/app/include" -L"/app/usr/local/include/aliflibwx/" -L"/app/include/wx-3.1-alif/wx/" -L"/lib" -L"/app/lib"  -lwx_gtk2u_alif_webview-3.1 -lwx_gtk2u_alif_propgrid-3.1 -lwx_gtk2u_alif_aui-3.1 -lwebkitgtk-1.0 -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lpangoft2-1.0 -lpango-1.0 -lfontconfig -lfreetype -lsoup-2.4 -lgio-2.0 -lgobject-2.0 -ljavascriptcoregtk-1.0 -lglib-2.0 -lwx_gtk2u_alif_stc-3.1 -lwx_gtk2u_alif_core-3.1 -lwx_baseu_alif-3.1 -lwxscintilla_alif-3.1 -lwxtiff_alif-3.1 -lwxjpeg_alif-3.1 -lwxpng_alif-3.1 -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lfontconfig -lfreetype -lgthread-2.0 -pthread -lglib-2.0 -lX11 -lXxf86vm -lSM -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfontconfig -lfreetype -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfontconfig -lfreetype -lwxzlib_alif-3.1 -lwxregexu_alif-3.1 -lwxexpat_alif-3.1 -pthread -m64 -static-libgcc -static-libstdc++ -ldl -lm -ldl -lm -O2;
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.yml:122]: nitpick: those should probably be upstream in a makefile or a proper buildsystem

File: org.aliflang.lang.metainfo.xml (hunk 3 above)
Comment (sulimanp) [reply org.aliflang.lang.metainfo.xml:8]: Done

File: org.aliflang.lang.yml (hunk 5 above)
Comment (sulimanp) [reply org.aliflang.lang.yml:9]: you are right

File: org.aliflang.lang.yml (hunk 6 above)
Comment (sulimanp) [reply org.aliflang.lang.yml:10]: To access home , I deleted it now

File: org.aliflang.lang.yml (hunk 7 above)
Comment (sulimanp) [reply org.aliflang.lang.yml:141]: If I install icon only in `/app/share/icons/hicolor/128x128/apps/` i get this error 
`Renaming org.aliflang.lang.metainfo.xml to share/appdata/org.aliflang.lang.appdata.xmlRunning appstream-composeProcessing application org.aliflang.langError loading desktop file: Failed to find icon: Failed to find icon org.aliflang.langError: ERROR: appstream-compose failed: Child process exited with code 1`

If i install icon only in `/app/usr/share/icons/hicolor/128x128/apps/` , he can't export app icon
`WARNING: Icon referenced in desktop file but not exported: org.aliflang.lang`


**I fixed it when installing the icon in both path**

File: org.aliflang.lang.yml [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+        url: https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2
+        sha256: 59c9b274bc451cf91a9ba1dd2c7fdcaf5d60b1b3aa83f2c9fa143417cc660722
+  
+  
+  - name: alif-lib
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/share/icons/hicolor/128x128/apps/ /app/usr/share/icons/hicolor/128x128/apps/ /app/share/{metainfo,applications} /app/usr/local/include/aliflibwx/ /app/usr/local/share/{aliflang,alifstudio} /app/usr/local/lib/aliflib/
+      - install ./usr/share/icons/hicolor/128x128/apps/org.aliflang.lang.png /app/share/icons/hicolor/128x128/apps/
+      - install ./usr/share/icons/hicolor/128x128/apps/org.aliflang.lang.png /app/usr/share/icons/hicolor/128x128/apps/
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.yml:93]:
```suggestion
```

File: org.aliflang.lang.yml [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+    sources:
+      - type: archive
+        url: https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2
+        sha256: 59c9b274bc451cf91a9ba1dd2c7fdcaf5d60b1b3aa83f2c9fa143417cc660722
+  
+  
+  - name: alif-lib
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/share/icons/hicolor/128x128/apps/ /app/usr/share/icons/hicolor/128x128/apps/ /app/share/{metainfo,applications} /app/usr/local/include/aliflibwx/ /app/usr/local/share/{aliflang,alifstudio} /app/usr/local/lib/aliflib/
```
Comment (bilelmoussaoui) [inline org.aliflang.lang.yml:91]:
```suggestion
      - mkdir -p /app/share/icons/hicolor/128x128/apps/ /app/share/{metainfo,applications} /app/usr/local/include/aliflibwx/ /app/usr/local/share/{aliflang,alifstudio} /app/usr/local/lib/aliflib/
```

Comment (sulimanp) [reply org.aliflang.lang.yml:91]: as I expected 
```
Removing files/bin/idle
Removing files/bin/gtk-update-icon-cache
Removing files/bin/gtk-query-immodules-2.0
Removing files/bin/gtk-demo
Removing files/bin/gtk-builder-convert
Removing files/bin/easy_install-2.7
Removing files/bin/easy_install
Removing files/bin/2to3
Removing files/bin
Renaming org.aliflang.lang.metainfo.xml to share/appdata/org.aliflang.lang.appdata.xml
Running appstream-compose
Processing application org.aliflang.lang
Error loading desktop file: Failed to find icon: Failed to find icon org.aliflang.lang
Error: ERROR: appstream-compose failed: Child process exited with code 1
----------------------------------------------------------------------------------------
~/git/flathub(org.aliflang.lang*) »                                      1 ↵ suliman@me

```
I fixed it when installing the icon in both paths

**Comment by sulimanp** [comment]
bot, build org.aliflang.lang

File: org.aliflang.lang.yml [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+      - type: git
+        url: https://github.com/wxWidgets/wxWidgets.git
+        commit: 152ec5103367d6d5e2c77ab180e9e432b58deac7
+
+
+  - name: boost
+    buildsystem: simple
+    build-commands:
+      - ./bootstrap.sh --prefix=/app --with-toolset=gcc --with-icu --with-python=/usr/bin/python2
+      - ./b2 install variant=release link=shared runtime-link=shared cxxflags="$CXXFLAGS" linkflags="$LDFLAGS" -j $FLATPAK_BUILDER_N_JOBS
```
Comment (barthalion) [inline org.aliflang.lang.yml:81]: Ideally you should limit it to libraries required by Alif instead of building full boost.

Comment (sulimanp) [reply org.aliflang.lang.yml:81]: Thanks for your feedback, it is really helpful

**Pull request closed without merging**